pkg/controller/machineset/machineset_controller_test.go

```diff
@@ -49,7 +49,18 @@ func TestReconcile(t *testing.T) {
 	if err := add(mgr, recFn, r.MachineToMachineSets); err != nil {
 		t.Errorf("error adding controller to manager: %v", err)
 	}
-	defer close(StartTestManager(mgr, t))
+
+	stop, errChan := StartTestManager(mgr, t)
+	defer close(stop)
+	defer func() {
+		select {
+		case err := <-errChan:
+			if err != nil {
+				t.Fatalf("error starting test manager: %v", err)
+			}
+		default:
+		}
+	}()
```

Comment:
This one needs updating to match the other one
```suggestion
	defer func() {
		close(stop)
		if err := <-errChan; err != nil {
			t.Fatalf("error starting test manager: %v", err)
		}
	}()
```